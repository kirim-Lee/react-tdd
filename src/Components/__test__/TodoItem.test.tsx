import { fireEvent, render, screen } from '@testing-library/react';
import 'jest-styled-components';

import { TodoItem } from '../';
import { MemoryRouter, useLocation } from 'react-router-dom';

describe('<TodoItem />', () => {
  it('렌더링이 잘 되는지 테스트', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/']}>
        <TodoItem id={1} label="default value" />
      </MemoryRouter>
    );

    const todoItem = screen.getByText('default value');

    expect(todoItem).toBeInTheDocument();

    const deleteButton = screen.getByText('삭제');

    expect(deleteButton).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('삭제버튼 테스트', () => {
    const handleClick = jest.fn();

    render(
      <MemoryRouter initialEntries={['/']}>
        <TodoItem id={1} label="default value" onDelete={handleClick} />
      </MemoryRouter>
    );

    const deleteButton = screen.getByText('삭제');

    fireEvent.click(deleteButton);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('라벨 클릭시 /detail 로 이동되는지 확인', () => {
    const TestComponent = () => {
      const location = useLocation();
      return (
        <div>
          <div>{location.pathname}</div>
          <TodoItem id={1} label="default value" />
        </div>
      );
    };

    render(
      <MemoryRouter initialEntries={['/']}>
        <TestComponent />
      </MemoryRouter>
    );

    const pathName = screen.getByText('/');
    expect(pathName).toBeInTheDocument();

    const todoItem = screen.getByText('default value');
    fireEvent.click(todoItem);
    expect(pathName.textContent).toBe('/detail/1');
  });
});
