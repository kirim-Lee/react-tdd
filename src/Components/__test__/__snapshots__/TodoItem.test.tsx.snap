// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItem /> 렌더링이 잘 되는지 테스트 1`] = `
.c2 {
  text-align: center;
  font-weight: bold;
  background-color: #d8b4fe;
  padding: 10px 20px;
  border-radius: 8px;
  border: 0;
}

.c2:hover {
  background-color: #4f46e5;
}

.c2:active {
  box-shadow: insect 5px 5px 10px rgba(0,0,0,0.1);
}

.c3 {
  color: #fff;
  font-size: 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 1px solid #bdbdbd;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 10px;
  padding: 10px;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 16px;
  margin-right: 20px;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #333;
}

<div>
  <div
    class="c0"
  >
    <a
      class="c1"
      href="/detail/1"
    >
      default value
    </a>
    <button
      class="c2"
    >
      <div
        class="c3"
      >
        삭제
      </div>
    </button>
  </div>
</div>
`;
